### DISCUSSION TITLE
Background index no longer works

### DISCUSSION THREAD

**Discussion by juxeii** [clangd/clangd#1131] [Q&A] [answered]
Hi,

I get this on the console
```
V[11:52:19.706] Background-index: Couldn't read /var/fpwork/reiss/bin/lib/clang/15.0.0/include/inttypes.h to validate stored index: No such file or directory
V[11:52:19.769] Background-index: Couldn't read /var/fpwork/reiss/bin/lib/clang/15.0.0/include/float.h to validate stored index: No such file or directory
V[11:52:19.836] Background-index: Couldn't read /var/fpwork/reiss/bin/lib/clang/15.0.0/include/xmmintrin.h to validate stored index: No such file or directory
V[11:52:19.915] Background-index: Couldn't read /var/fpwork/reiss/bin/lib/clang/15.0.0/include/unwind.h to validate stored index: No such file or directory
V[11:52:19.921] Background-index: Couldn't read /var/fpwork/reiss/bin/lib/clang/15.0.0/include/stdint.h to validate store
[log.txt](https://github.com/clangd/clangd/files/8581926/log.txt)
d index: No such file or directory
V[11:52:20.120] Background-index: Couldn't read /var/fpwork/reiss/bin/lib/clang/15.0.0/include/limits.h to validate stored index: No such file or directory
V[11:52:20.152] Background-index: Couldn't read /var/fpwork/reiss/bin/lib/clang/15.0.0/include/mmintrin.h to validate stored index: No such file or directory
V[11:52:20.163] Background-index: Couldn't read /var/fpwork/reiss/bin/lib/clang/15.0.0/include/stddef.h to validate stored index: No such file or directory
V[11:52:20.175] Background-index: Couldn't read /var/fpwork/reiss/bin/lib/clang/15.0.0/include/emmintrin.h to validate stored index: No such file or directory
V[11:52:20.613] Background-index: Couldn't read /var/fpwork/reiss/bin/lib/clang/15.0.0/include/__stddef_max_align_t.h to validate stored index: No such file or directory
V[11:52:20.614] Background-index: Couldn't read /var/fpwork/reiss/bin/lib/clang/15.0.0/include/stdarg.h to validate stored index: No such file or directory
V[11:52:20.693] Background-index: Couldn't read /var/fpwork/reiss/bin/lib/clang/15.0.0/include/mm_malloc.h to validate stored index: No such file or directory
```
Where does this path `lib/clang/15.0.0/include` come from?
How can I fix this, since it worked before.

EDIT: I could somehow fix this, in moving the folder structure to the needed directory.
But now, it seems that the backgroung is still not working. Previously there were ~10000 files to be indexed. According to logs, it only idexed two files(log file attache).
There is also no longer a `.cache` folder in my project directory, where previously there was one.

**Comment by juxeii** [comment]
[log.txt](https://github.com/clangd/clangd/files/8581929/log.txt)

**Comment by juxeii** [comment, marked as answer]
Ok, I just figured, that the .cache folder is always right beside compile_commands.json. I had it previously in some other location.

So, the only question remaining is where this folder "lib/clang/15.0.0/include" stems from.

**Reply by sam-mccall** [reply]
That's the built-in headers directory - it looks like you had clangd installed there when you built the index, and now you have it installed somewhere else?

**Reply by juxeii** [reply]
Yes.

**Reply by sam-mccall** [reply]
Those log messages aren't harmful/errors, they're just logging something slightly unexpected. If things are otherwise working, don't worry about them.

**Reply by juxeii** [reply]
Well, I worried, because I had no index cache where I expected it to be.
Was just surprised that it did work anyway. I wondered if you guys implemented a new feature which no longer needed a cache ;)
